### ISSUE TITLE
Android random crash

### ISSUE THREAD

**Issue by D10S60948** [transistorsoft/react-native-background-geolocation#1433]
## Your Environment
* Plugin version: 4.3.0
* Platform: Android
* OS version: 10, 11
* Device manufacturer / model: POCOPHONE F1 (QKQ1.190828.002), SM-A125F (QP1A.190711.020)
* React Native version (`react-native -v`): 0.66.4

```javascript <!-- Syntax-highlighting:  paste your code below -->
    const startRecording = async () => {
        BackgroundGeolocation.ready({
            desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
            distanceFilter: 10,
            stopTimeout: 10,
            debug: false,
            logLevel: BackgroundGeolocation.LOG_LEVEL_OFF,
            stopOnTerminate: true,
            startOnBoot: false,
            allowIdenticalLocations: true,
            activityType: BackgroundGeolocation.ACTIVITY_TYPE_FITNESS,
            autoSync: false,
            isMoving: true,
            maxRecordsToPersist: 10000,
        }, async (state) => {
            console.log("- BackgroundGeolocation is configured and ready: ", state.enabled);
            if (!state.enabled) {
                await BackgroundGeolocation.start(() => {
                    console.log("- Start success");
                    startLocationActivity();
                });
            }
            else {
                startLocationActivity();
            }
        });
    };

    const startLocationActivity = () => {
        BackgroundGeolocation.changePace(true);
        startTrackingLocation();
    }    

    const stopRecording = useCallback(async () => {
        console.log('stopRecording');
        await BackgroundGeolocation.changePace(false);
        BackgroundGeolocation.removeAllListeners();
    }, []);
```

## Expected Behavior

App should not crash
## Actual Behavior

App does crash

## Steps to Reproduce
It happens randomly, caught with Sentry. no pattern I recognized.
reproduced by mistake so caught the log.

## Context
Nothing special. It's a cycling app and it crashes even out of the `activity record` screen (e.g. plugin is not in use).

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
01-10 20:17:51.046 29741 30716 E AndroidRuntime: FATAL EXCEPTION: androidmapsapi-Snapshot
01-10 20:17:51.046 29741 30716 E AndroidRuntime: Process: com.xxx.xxx., PID: 29741
01-10 20:17:51.046 29741 30716 E AndroidRuntime: java.lang.IllegalArgumentException: width and height must be > 0
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:1113)
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:1080)
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:1030)
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:991)
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at com.google.maps.api.android.lib6.impl.a.d(:com.google.android.gms.dynamite_mapsdynamite@214815088@21.48.15 (150800-0):2)
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at com.google.maps.api.android.lib6.impl.bl.run(:com.google.android.gms.dynamite_mapsdynamite@214815088@21.48.15 (150800-0):1)
01-10 20:17:51.046 29741 30716 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:923)
01-10 20:17:51.154 29741 30716 E TSLocationManager: [c.t.l.a.BackgroundGeolocation$x0 uncaughtException] 
01-10 20:17:51.154 29741 30716 E TSLocationManager:   ‼️  Uncaught Exception: width and height must be > 0
01-10 20:17:51.154 29741 30716 E TSLocationManager: {"activityRecognitionInterval":10000,"allowIdenticalLocations":true,"authorization":{},"autoSync":false,"autoSyncThreshold":0,"backgroundPermissionRationale":{"title":"Allow {applicationName} to access this device's location even when closed or not in use?","message":"[CHANGEME] This app collects location data for FEATURE X and FEATURE Y.","positiveAction":"Change to \"{backgroundPermissionOptionLabel}\"","negativeAction":""},"batchSync":false,"configUrl":"","debug":false,"deferTime":0,"desiredAccuracy":-1,"desiredOdometerAccuracy":100,"disableAutoSyncOnCellular":false,"disableElasticity":false,"disableLocationAuthorizationAlert":false,"disableMotionActivityUpdates":false,"disableStopDetection":false,"distanceFilter":10,"elasticityMultiplier":1,"enableHeadless":false,"enableTimestampMeta":false,"extras":{},"fastestLocationUpdateInterval":-1,"foregroundService":true,"geofenceInitialTriggerEntry":true,"geofenceModeHighAccuracy":false,"geofenceProximityRadius":1000,"geofenceTemplate":"","headers":{},"headlessJobService":"com.transistorsoft.rnbackgroundgeolocation.HeadlessTask","heartbeatInterval":-1,"httpRootProperty":"location","httpTimeout":60000,"isMoving":false,"locationAuthorizationRequest":"Always","locationTemplate":"","locationTimeout":60,"locationUpdateInterval":1000,"locationsOrderDirection":"ASC","logLevel":0,"logMaxDays":3,"maxBatchSize":-1,"maxDaysToPersist":1,"maxRecordsToPersist":10000,"method":"POST","minimumActivityRecognitionConfidence":75,"motionTriggerDelay":0,"notification":{"layout":"","title":"","text":"Location Service activated","color":"","channelName":"TSLocationManager","smallIcon":"","largeIcon":"","priority":0,"sticky":false,"strings":{},"actions":[]},"params":{},"persist":true,"persistMode":2,"schedule":[],"scheduleUseAlarmManager":false,"speedJumpFilter":300,"startOnBoot":false,"stationaryRadius":25,"stopAfterElapsedMinutes":0,"stopOnStationary":false,"stopOnTerminate":true,"stopTimeout":10,"triggerActivities":"in_vehicle, on_bicycle, on_foot, running, walking","url":"","useSignificantChangesOnly":false,"enabled":true,"schedulerEnabled":false,"trackingMode":1,"odometer":93282.984375,"isFirstBoot":false,"didLaunchInBackground":false,"didDeviceReboot":false}
01-10 20:17:51.154 29741 30716 E TSLocationManager: java.lang.IllegalArgumentException: width and height must be > 0
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:1113)
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:1080)
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:1030)
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at android.graphics.Bitmap.createBitmap(Bitmap.java:991)
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at com.google.maps.api.android.lib6.impl.a.d(:com.google.android.gms.dynamite_mapsdynamite@214815088@21.48.15 (150800-0):2)
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at com.google.maps.api.android.lib6.impl.bl.run(:com.google.android.gms.dynamite_mapsdynamite@214815088@21.48.15 (150800-0):1)
01-10 20:17:51.154 29741 30716 E TSLocationManager: 	at java.lang.Thread.run(Thread.java:923)
```

</details>

**Comment by christocracy** [comment]
Your crash is not *caused by* background-geolocation the plugin simply has a "catch all" exception-handler so that uncaught exceptions can be logged into the plugin's log database. 

Your error seems to be caused by some google maps code, likely related to an invalid map marker image, or something (since there are complaints about a bitmap size being invalid).

```
at com.google.maps
```

**Comment by D10S60948** [comment]
I see. Thanks for the explanation.

**Comment by markrickert** [comment]
I came here because i was also encountering the error message: `java.lang.IllegalArgumentException: width and height must be > 0`.

There were no similar issues on the react-native-maps repo but i finally figured out that for some reason when navigating (with react-navigation) to a modal screen from a screen that has a MapView on it, android was crashing.

<pre>react-native: 0.73.3
react-native-maps: 1.10.0
react-native-background-geolocation: 4.14.6</pre>

What I did to solve it was:

```tsx
const isFocused = useIsFocused(); // from react-navigation

return (
  <>
    {isFocused && <MapView />}
  </>
)
```
I'm posting this solution here in the hopes that it will help others that may run across the same thing since this was one of the only google search results for this particular error message. This crash had nothing to do with `react-native-background-geolocation`
